File: src/main/java/org/stt/Strings.java
package org.stt;

import java.util.Objects;

public class Strings {
    private Strings() {
    }

    public static String commonPrefix(String a, String b) {
        Objects.requireNonNull(a);
        Objects.requireNonNull(b);
        for (int i = 0; i < a.length() && i < b.length(); i++) {
            if (a.charAt(i) != b.charAt(i)) {
                return a.substring(0, i);
            }
        }
        return a;
    }
}

File: src/main/kotlin/org/stt/Strings.kt
package org.stt

object Strings {

    fun commonPrefix(a: String, b: String): String {
        var i = 0
        while (i < a.length && i < b.length) {
            if (a[i] != b[i]) {
                return a.substring(0, i)
            }
            i++
        }
        return a
    }
}
